R/Vietnam.R

```diff
@@ -0,0 +1,124 @@
+#' region$return()
+#' }
+Vietnam <- R6::R6Class("Vietnam",
+                       inherit = DataClass,
+                       public = list(
+                         
+                         # Core Attributes (amend each paramater for country specific infomation)
+                         #' @field origin name of country to fetch data for
+                         origin = "Vietnam",
+                         #' @field supported_levels List of supported levels.
+                         supported_levels = list("1"),
+                         #' @field supported_region_names List of region names in order of level.
+                         supported_region_names = list("1" = "region"),
+                         #' @field supported_region_codes List of region codes in order of level.
+                         supported_region_codes = list("1" = "is_3166_2"),
+                         #' @field common_data_urls List of named links to raw data.
+                         # nolint start
+                         common_data_urls = list(
+                           "case_by_time" = 'https://covid.ncsc.gov.vn/api/v3/covid/provinces?filter_type=case_by_time',
+                           "death_by_time" = 'https://covid.ncsc.gov.vn/api/v3/covid/provinces?filter_type=death_by_time',
+                           "recovered_by_time" = 'https://covid.ncsc.gov.vn/api/v3/covid/provinces?filter_type=recovered_by_time'
+                           
+                         ),
+                         # nolint end
+                         #' @field source_data_cols existing columns within the raw data
+                         source_data_cols = c(
+                           "cases_total", "deaths_total", "recovered_total"
+                         ),
+                         #' @field source_text Plain text description of the source of the data
+                         source_text = "Public COVID-19 data curated by 5F team",
+                         #' @field source_url Website address for explanation/introduction of the
+                         #' data
+                         source_url = "https://covid.ncsc.gov.vn", # nolint
+                         
+                         #' @description Set up a table of region codes for clean data
+                         #' @importFrom tibble tibble
+                         set_region_codes = function(){
+                           self$codes_lookup$`1` <- covidregionaldata::vietnam_codes
+                         },
+                         
+                         #' @description Provincial Level Data
+                         #' cleaning
+                         #' @param ... pass additional arguments
+                         #'
+                         #' @importFrom dplyr filter select mutate rename
+                         #' @importFrom tidyr replace_na drop_na
+                         #' @importFrom lubridate dmy 
+                         #' @importFrom jsonlite fromJSON
```

Comment:
I've checked and `httr` imports `jsonlite` so we're already getting `jsonlite` but I think we should probably import it explicitly by adding it to the DESCRIPTION file.

Reply:
Added jsonlite to `DESCRIPTION` and reran `roxygenize`.